### PR TITLE
Attempt to diagnose arguments to `-sdk` that point to unsupported SDKs

### PR DESCRIPTION
**Pull request by artemcm**
Resolves rdar://70215624

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci please test

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 1]
```diff
@@ -2005,8 +2005,11 @@ extension Driver {
 
       if !fileSystem.exists(path) {
         diagnosticsEngine.emit(.warning_no_such_sdk(sdkPath))
+      } else if targetTriple != nil,
+                isSDKTooOld(sdkPath: path, target: targetTriple!) {
```
Comment (cltnschlosser) [inline Sources/SwiftDriver/Driver/Driver.swift:2009]:
```suggestion
      } else if let targetTriple = targetTriple,
                isSDKTooOld(sdkPath: path, target: targetTriple) {
```
Don't have to force unwrap here if you use `if let`

**Comment by nkcsgexi** [comment]
Can we use the `DarwinSDKInfo` for checks like this? We may use an SDK path that's without the version number in it (a symlink to the actual SDK dir).

**Comment by artemcm** [comment]
> Can we use the `DarwinSDKInfo` for checks like this? We may use an SDK path that's without the version number in it (a symlink to the actual SDK dir).

Oh, I hadn't realized we already had this implemented. Great, I'll look into that.

File: Sources/SwiftDriver/Driver/Driver.swift (hunk 1 above)
Comment (artemcm) [reply Sources/SwiftDriver/Driver/Driver.swift]: This code looks a bit different now, but thank you for taking a look.

**Comment by artemcm** [comment]
> Can we use the `DarwinSDKInfo` for checks like this? We may use an SDK path that's without the version number in it (a symlink to the actual SDK dir).

@nkcsgexi, I changed the approach to use the `DarwinSDKInfo`.

**Comment by artemcm** [comment]
@swift-ci please test

File: TestInputs/SDKChecks/watchOS3.0.sdk/SDKSettings.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+	"Version": "3.0",
+	"VersionMap": {
+		"macOS_iOSMac": {},
+		"iOSMac_macOS": {}
+	},
```
Comment (nkcsgexi) [inline TestInputs/SDKChecks/watchOS3.0.sdk/SDKSettings.json:6]: Thanks! I think we shouldn't need `VersionMap` in non-macosx SDKs.

Comment (artemcm) [reply TestInputs/SDKChecks/watchOS3.0.sdk/SDKSettings.json]: Good catch, thanks. 👍🏼

**Comment by artemcm** [comment]
@swift-ci please test

**Review by nkcsgexi — APPROVED** [review]
LGTM!

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 3]
```diff
@@ -2015,6 +2018,32 @@ extension Driver {
...
+    guard let sdkVersion = Version(potentiallyIncompleteVersionString: sdkInfo.versionString) else {
+      diagnosticsEngine.emit(.warning_fail_parse_sdk_ver(sdkInfo.versionString, sdkPath.pathString))
+      return false
+    }
+    if sdkInfo.canonicalName.hasPrefix("macos") {
+      return sdkVersion < Version(10, 15, 0)
+    } else if sdkInfo.canonicalName.hasPrefix("iphoneos") ||
+                sdkInfo.canonicalName.hasPrefix("appletvos") {
+      return sdkVersion < Version(13, 0, 0)
+    } else if sdkInfo.canonicalName.hasPrefix("watchos") {
```
Comment (nkcsgexi) [inline Sources/SwiftDriver/Driver/Driver.swift:2039]: Sorry about missing this. Do we need to handle simulator SDKs here?

Comment (nkcsgexi) [reply Sources/SwiftDriver/Driver/Driver.swift:2039]: Thanks! Using a more concise prefix will just work.

**Pull request merged**